repository/diet_repository.py

```diff
@@ -0,0 +1,57 @@
+from domain.entity.diet_requirement import DietRequirement
+
+
+def save_dietary_requirements(session, user_id, dietary_requirements):
+    """
+    A helper function for upload the data to the database.
+    This function will map the argument from the frontend with the statement in the database
+    Then using the session to update the database
+    This function is allowed to change the dietary requirement for the user by providing the user_id
+
+    Parameters:
+        session: A SQLAlchemy Session for the query.
+        user_id: The user_id to identify create a new dietary requirement or
+                change the current dietary requirement in the database
+        dietary_requirements: The parser contains the data required for update
+
+    Returns:
+        True if the data is updated; False if the data is not upload
+    """
+    # Change the diet requirement in the database
+    diet = session.query(DietRequirement).filter(DietRequirement.user_id == user_id).first()
+
+    if not diet:
+        # Create a new diet requirement object if none exists
+        diet = DietRequirement(user_id=user_id)
```

Comment:
I've been assuming that the intended behaviour of this function is to overwrite any existing requirements. But, if we are creating an object from existing entries and then updating the fields on that object, would the fields already stored as True remain True? At this point I don't really have a preference, but we need to make the behaviour clear in both tests and documentation.

Reply:
Thank you so much for your suggestion. The reason I didn't create a new dietary requirement without checking because we excepting each user to have only one dietary requirement in the table. You have a good point that I haven't tested whether the selection stored in True will be able to switch back to False. I will add two testing cases to test this situation. And I will find a solution to solve this problem.